tapioca/tapioca.py

```diff
@@ -170,14 +170,24 @@ def _get_iterator_next_request_kwargs(self):
         return self._api.get_iterator_next_request_kwargs(
             self._request_kwargs, self._data, self._response)
 
+    def _has_next(self, actual, kwargs):
+        has_more_page = "max_pages" not in kwargs or kwargs["max_pages"] > actual["page"]
+        has_more_items = "max_items" not in kwargs or kwargs["max_items"] > actual["items"]
+        return  has_more_page and has_more_items
+
     def pages(self, **kwargs):
         executor = self
         iterator_list = executor._get_iterator_list()
+        actual = {"page": 0, "items": 0}
 
-        while iterator_list:
+        while iterator_list and self._has_next(actual, kwargs):
             for item in iterator_list:
                 yield self._wrap_in_tapioca(item)
-
+                actual["items"] += 1
+                if not self._has_next(actual, kwargs):
```

Comment:
it's more redable if you make the item verification right here:
``if max_items and item_count < max_items:``